python-package/setup.py

```diff
@@ -129,15 +116,15 @@ def compile_cpp(
     bit32: bool = False,
     integrated_opencl: bool = False
 ) -> None:
-
-    if os.path.exists(os.path.join(CURRENT_DIR, "build_cpp")):
-        remove_tree(os.path.join(CURRENT_DIR, "build_cpp"))
-    os.makedirs(os.path.join(CURRENT_DIR, "build_cpp"))
-    os.chdir(os.path.join(CURRENT_DIR, "build_cpp"))
+    build_dir = CURRENT_DIR / "build_cpp"
+    rmtree(build_dir, ignore_errors=True)
+    build_dir.mkdir(parents=True)
+    original_dir = Path.cwd()
+    chdir(build_dir)
```

Comment:
Unfortunately, we have to `chdir` due to old CMake doesn't support specifying building directory (I don't want to use undocumented flag):
https://stackoverflow.com/a/24435795
https://stackoverflow.com/a/20611964

> ` cmake [<options>] -S <path-to-source> -B <path-to-build>`
   https://cmake.org/cmake/help/latest/manual/cmake.1.html#synopsis

and no such possibility in CMake < 3.13
https://cmake.org/cmake/help/v3.12/manual/cmake.1.html#synopsis

Reply:
thanks for calling this out, makes sense!